src/main/java/com/opentok/util/HttpClient.java

```diff
@@ -380,7 +386,66 @@ public String setArchiveLayout(String archiveId, ArchiveProperties properties) t
                             " response code: " + response.getStatusCode());
             }
         } catch (InterruptedException | ExecutionException e) {
-            throw new RequestException("Could not delete an OpenTok Archive. archiveId = " + archiveId, e);
+            throw new RequestException("Could not set the layout, archiveId = " + archiveId, e);
+        }
+        return responseString;
+    }
+
+    public String setArchiveStreamLayout(String sessionId, StreamListProperties properties) throws OpenTokException {
+        String responseString = null;
+        char doubleQuotes = '"';
+        String url = this.apiUrl + "/v2/project/" + this.apiKey + "/session/" + sessionId + "/stream";
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try {
+            JsonFactory factory = new JsonFactory();
+            //Using JsonGenerator as layoutClassList values must be in double quotes and ObjectMapper
+            // adds extra escape characters
+            JsonGenerator jGenerator = factory.createGenerator(outputStream);
+            jGenerator.writeStartObject();
+            jGenerator.writeArrayFieldStart("items");
+
+            for(StreamProperties stream : properties.getStreamList()) {
+                jGenerator.writeStartObject();
+                jGenerator.writeFieldName("id");
+                jGenerator.writeString(stream.id());
+                jGenerator.writeFieldName("layoutClassList");
+                List<String> stringList = stream.getLayoutClass();
+                StringJoiner sj = new StringJoiner(",");
+                stringList.stream().forEach(e -> sj.add(doubleQuotes + e + doubleQuotes));
+                jGenerator.writeRawValue("["+ sj.toString() + "]");
```

Comment:
Will this work with an empty layout class list (to clear the classes for a stream)? In other words, if you do not call the `StreamProperties.Builder.addLayoutClass()` method when building the StreamProperties object, the stream's layout class list should be cleared.

Reply:
At present the code does
<code>
{"items":[
  {
    "id":"STREAMID1",
    "layoutClassList":[]
  }
  ]
}
</code>
and tested that it works